great_expectations/core/expectation_suite.py

```diff
@@ -739,22 +767,22 @@ def add_expectation(
 
         # noqa: DAR402
         """
-        self._validate_expectation_configuration_before_adding(
-            expectation_configuration
-        )
+        self._build_expectation(expectation_configuration)
         return self._add_expectation(
             expectation_configuration=expectation_configuration,
             send_usage_event=send_usage_event,
             match_type=match_type,
             overwrite_existing=overwrite_existing,
         )
 
-    def _validate_expectation_configuration_before_adding(
+    def _build_expectation(
         self, expectation_configuration: ExpectationConfiguration
-    ):
+    ) -> Expectation:
         try:
             class_ = get_expectation_impl(expectation_configuration.expectation_type)
-            _ = class_(expectation_configuration)  # Implicitly validates in constructor
+            expectation = class_(configuration=expectation_configuration)
+            expectation.validate_configuration(expectation_configuration)
```

Comment:
`validate_configuration` is already called in the constructor of the Expectation (this validation will eventually be done by Pydantic) - we don't need L784 here we can just return the Expectation